tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedReadHandleImplV2.java

```diff
@@ -233,8 +233,8 @@ public CompletableFuture<LedgerEntries> readAsync(long firstEntry, long lastEntr
                     entries.forEach(LedgerEntry::close);
                 }
 
-                promise.complete(LedgerEntriesImpl.create(entries));
```

Comment:
This seems to be a critical bug. Do we have any test to cover this logic? (Seems not). maybe we should ask the code author to add a test for this logic. (as a separate PR).

Reply:
There's an existing test BlobStoreManagedLedgerOffloaderStreamingTest.testReadAndWriteAcrossSegment to cover this. It fails for me consistently when running on command line locally. It seems that it requires a fast CPU to reproduce and this test has rarely failed in CI.

Reply:
@heesung-sn It looks like this code never went to production, PIP-76 hasn't been completed. I added details to the description.